File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/9 - 20 and 21 April 2019/05. Easter Bake.py
import math
import sys
easter_bread = int(input())

total_sugar = 0
total_flour = 0

max_flour = -sys.maxsize - 1
max_sugar = -sys.maxsize - 1

for i in range(easter_bread):
    sugar = int(input())
    total_sugar += sugar
    if sugar > max_sugar:
        max_sugar = sugar
    flour = int(input())
    total_flour += flour
    if flour > max_flour:
        max_flour = flour


print(f"Sugar: {math.ceil(total_sugar / 950)}")
print(f"Flour: {math.ceil(total_flour / 750)}")
print(f"Max used flour is {max_flour} grams, max used sugar is {max_sugar} grams.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/01. Online Exam - 20 and 21 April 2019/05. Easter Bake.js
function easterBake(input) {
    // Mask
    let count = Number(input.shift());

    let maxFlour = 0;
    let maxSugar = 0;

    let totalSugar = 0;
    let totalFlour = 0;

    for (let bread = 1; bread <= count; bread += 1) {
        let sugar = Number(input.shift());
        if (sugar > maxSugar) {maxSugar = sugar;}
        totalSugar += sugar;

        let flour = Number(input.shift());
        if (flour > maxFlour) {maxFlour = flour;}
        totalFlour += flour;
    }

    console.log(`Sugar: ${Math.ceil(totalSugar / 950)}`);
    console.log(`Flour: ${Math.ceil(totalFlour / 750)}`);
    console.log(`Max used flour is ${maxFlour} grams, max used sugar is ${maxSugar} grams.`);
}

easterBake([3, 400, 350, 250, 300, 450, 380,]);  // Should return:
// Sugar: 2
// Flour: 2
// Max used flour is 380 grams, max used sugar is 450 grams.

easterBake([4, 500, 350, 560, 430, 600, 345, 578, 543]);  // Should return:
// Sugar: 3
// Flour: 3
// Max used flour is 543 grams, max used sugar is 600 grams.
